crates/sui/src/zklogin_commands_util.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use fastcrypto::traits::EncodeDecodeBase64;
+use fastcrypto_zkp::bn254::utils::get_enoki_address;
+use fastcrypto_zkp::bn254::zk_login::ZkLoginInputs;
+use regex::Regex;
+use reqwest::Client;
+use serde::Deserialize;
+use serde_json::json;
+use shared_crypto::intent::Intent;
+use std::io;
+use std::io::Write;
+use sui_json_rpc_types::SuiTransactionBlockResponseOptions;
+use sui_keys::keystore::{AccountKeystore, Keystore};
+use sui_sdk::SuiClientBuilder;
+use sui_types::base_types::SuiAddress;
+use sui_types::committee::EpochId;
+use sui_types::signature::GenericSignature;
+use sui_types::transaction::Transaction;
+use sui_types::zk_login_authenticator::ZkLoginAuthenticator;
+
+const GAS_URL: &str = "http://127.0.0.1:9123/gas";
+const SALT_SERVER_URL: &str = "http://salt.api.mystenlabs.com/get_salt";
+const PROVER_SERVER_URL: &str = "http://185.209.177.123:8000/zkp";
+
+/// Read a line from stdin, parse the id_token field and return.
+pub fn read_cli_line() -> Result<String, anyhow::Error> {
+    let mut s = String::new();
+    let _ = io::stdout().flush();
+    io::stdin().read_line(&mut s)?;
+    let full_url = s.trim_end().to_string();
+    let mut parsed_token = "";
+    let re = Regex::new(r"id_token=([^&]+)").unwrap();
+    if let Some(captures) = re.captures(&full_url) {
+        if let Some(id_token) = captures.get(1) {
+            parsed_token = id_token.as_str();
+        }
+    }
+    Ok(parsed_token.to_string())
+}
+
+/// A util function to request gas token from faucet for the given address.
+pub(crate) async fn request_tokens_from_faucet(address: SuiAddress) -> Result<(), anyhow::Error> {
+    let client = Client::new();
+    client
+        .post(GAS_URL)
+        .header("Content-Type", "application/json")
+        .json(&json![{
+            "FixedAmountRequest": {
+                "recipient": &address.to_string()
+            }
+        }])
+        .send()
+        .await?;
+    Ok(())
+}
+
+/// Call the salt server to get the salt based on the given JWT token.
+pub async fn get_salt(jwt_token: &str) -> Result<String, anyhow::Error> {
```

Comment:
this should live in an sdk - for a lack of it im putting it here for now. same for get_proof